spec/factories/book.rb

```diff
@@ -0,0 +1,12 @@
+require 'faker'
+
+FactoryBot.define do
+  factory :book do
+    genre { Faker::Book.genre }
+    title { Faker::Book.title }
+    author { Faker::Name.name }
+    year { Faker::Number.between(from: 1900, to: 2019) }
```

Comment:
what happens if the year of the  book is greater than the actual year? you should add a validation in the model something like this
```
  validates :year_validation
  def year_validation
    add_errors(:wrong_year, 'el año es de una fecha que no corresponde')year <= Time.zone.today.year
  end
```